client/components/domains/register-domain-step/index.jsx

```diff
@@ -220,7 +223,7 @@ var RegisterDomainStep = React.createClass( {
 		async.parallel(
 			[
 				callback => {
-					if ( domain.indexOf( '.' ) < 0 ) {
+					if ( ! domain.match( /.{3,}\..{2,}/ ) ) {
```

Comment:
I think `.` matches a dot right as well right?
Maybe something like `/\w{3,}\.\w{2,}/` would be a tiny bit better.